### PR TITLE
Operations manager - update list scheduler finished op method

### PR DESCRIPTION
**Pull request by mflilian**
### What ❓ 

1. Added a new signature to the Method ListSchedulerFinishedOperations;
2. New arguments: page and pageSize, page will be indexed as array (starts at zero);
3. New return paremeter: int, represents the total number of finishedOperations.

### Why 🤔 
Given the need to paginate the Finished Operations list, we have to make additional updates on Operations manager.

### CODE REVIEW

File: internal/core/services/operation_manager/operation_manager.go [hunk 1]
```diff
@@ -187,9 +187,14 @@ func (om *OperationManager) ListSchedulerActiveOperations(ctx context.Context, s
 	return ops, nil
 }
 
-func (om *OperationManager) ListSchedulerFinishedOperations(ctx context.Context, schedulerName string) ([]*operation.Operation, error) {
-	operations, _, err := om.Storage.ListSchedulerFinishedOperations(ctx, schedulerName, 0, -1)
-	return operations, err
+func (om *OperationManager) ListSchedulerFinishedOperations(ctx context.Context, schedulerName string, page, pageSize int64) (result []*operation.Operation, total int64, err error) {
+
```
Comment (guilhermocc) [inline internal/core/services/operation_manager/operation_manager.go:191]:
```suggestion

```

File: internal/core/services/operation_manager/operation_manager.go [hunk 2]
```diff
@@ -187,9 +187,14 @@ func (om *OperationManager) ListSchedulerActiveOperations(ctx context.Context, s
...
+	result, total, err = om.Storage.ListSchedulerFinishedOperations(ctx, schedulerName, page, pageSize)
+	if err != nil {
+		return nil, 0, fmt.Errorf("failed to list scheduler finished operations for scheduler %s : %w", schedulerName, err)
```
Comment (guilhermocc) [inline internal/core/services/operation_manager/operation_manager.go:194]:
```suggestion
		return nil, 0, fmt.Errorf("failed to list finished operations for scheduler %s : %w", schedulerName, err)
```

File: internal/core/services/operation_manager/operation_manager_test.go [hunk 3]
```diff
@@ -617,7 +617,7 @@ func TestListSchedulerActiveOperations(t *testing.T) {
 }
 
 func TestListSchedulerFinishedOperations(t *testing.T) {
-	t.Run("it returns an operation list with finished status", func(t *testing.T) {
+	t.Run("Return an operation list with finished status and its total", func(t *testing.T) {
```
Comment (guilhermocc) [inline internal/core/services/operation_manager/operation_manager_test.go:620]:
```suggestion
	t.Run("return an operation list with finished status and its total", func(t *testing.T) {
```
@mflilian We are not using Capital letters for test cases here in Maestro

File: internal/core/services/operation_manager/operation_manager_test.go [hunk 4]
```diff
@@ -636,12 +636,36 @@ func TestListSchedulerFinishedOperations(t *testing.T) {
...
 
-		operationStorage.EXPECT().ListSchedulerFinishedOperations(ctx, schedulerName, int64(0), int64(-1)).Return(operationsResult, int64(0), nil)
-		operations, err := opManager.ListSchedulerFinishedOperations(ctx, schedulerName)
+		operationStorage.EXPECT().ListSchedulerFinishedOperations(ctx, schedulerName, page, pageSize).Return(operationsResult, int64(3), nil)
+		operations, total, err := opManager.ListSchedulerFinishedOperations(ctx, schedulerName, page, pageSize)
 		require.NoError(t, err)
+		assert.Equal(t, int64(3), total)
 		require.ElementsMatch(t, operationsResult, operations)
 	})
+	t.Run("Return error when an error occurs in operation storage", func(t *testing.T) {
```
Comment (guilhermocc) [inline internal/core/services/operation_manager/operation_manager_test.go:648]:
```suggestion
	t.Run("return error when some error occurs in operation storage", func(t *testing.T) {
```

File: proto/apidocs.swagger.json [hunk 5]
```diff
@@ -103,23 +103,7 @@
...
-                  "description": "Indicates the room status."
-                },
-                "timestamp": {
-                  "type": "string",
-                  "format": "int64",
-                  "description": "Timestamp of the ping event."
-                }
-              },
-              "description": "The ping request."
+              "$ref": "#/definitions/v1UpdateRoomWithPingRequest"
```
Comment (guilhermocc) [inline proto/apidocs.swagger.json:106]: @mflilian I guess this file change should not be included in this PR, right?

**Review by guilhermocc — APPROVED** [review]

File: proto/apidocs.swagger.json (hunk 5 above)
Comment (mflilian) [reply proto/apidocs.swagger.json:106]: YESS!!!!!!

Comment (mflilian) [reply proto/apidocs.swagger.json:106]: I'm gonna remove it, thanks Gui

File: internal/core/services/operation_manager/operation_manager.go (hunk 1 above)
Comment (mflilian) [reply internal/core/services/operation_manager/operation_manager.go]: ✅

File: internal/core/services/operation_manager/operation_manager.go (hunk 2 above)
Comment (mflilian) [reply internal/core/services/operation_manager/operation_manager.go]: ✅

File: internal/core/services/operation_manager/operation_manager_test.go (hunk 3 above)
Comment (mflilian) [reply internal/core/services/operation_manager/operation_manager_test.go]: thanks, Gui!
